### PR TITLE
[autoscaler] Add option of returning node metadata from non_terminated_nodes.

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

Returning nothing from a terminate nodes request feels squishy to me.
Mind if I add the option of returning deleted node metadata? 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/node_provider.py [hunk 1]
```diff
@@ -125,8 +125,11 @@ def terminate_node(self, node_id: str) -> None:
         """Terminates the specified node."""
         raise NotImplementedError
 
-    def terminate_nodes(self, node_ids: List[str]) -> None:
-        """Terminates a set of nodes. May be overridden with a batch method."""
+    def terminate_nodes(self, node_ids: List[str]) -> Optional[Dict[str, Any]]:
+        """Terminates a set of nodes. May be overridden with a batch method,
+        which optionally may return a mapping from deleted node ids to node
+        metadata.
+        """
```
Comment (ijrsvt) [inline python/ray/autoscaler/node_provider.py:132]: Should we also (optionally) return this mapping from `terminate_node`?

**Review by ijrsvt — COMMENTED** [review]
@DmitriGekhtman My only question is how will the output be used here? I.e. are we going to have the autoscaler doing something with this output, or is it more used as confirmation that the autoscaler may print out?

**Comment by DmitriGekhtman** [comment]
> @DmitriGekhtman My only question is how will the output be used here? I.e. are we going to have the autoscaler doing something with this output, or is it more used as confirmation that the autoscaler may print out?

Just for confirmation -- right now, I find it convenient to have for testing.
In particular for an RPC NodeProvider, it is useful to hear back from the server "I have done the correct thing with these nodes, here's [some info] about what has happened."
Don't know exactly what [some info] should be right now (besides in test scenarios). I'm mostly just uncomfortable sending RPCs into black holes.

File: python/ray/autoscaler/node_provider.py (hunk 1 above)
Comment (DmitriGekhtman) [reply python/ray/autoscaler/node_provider.py:132]: ehh...yeah, had better as some node providers define terminate_node(node) as
terminate_nodes([node])

**Review by mwtian — APPROVED** [review]

**Review by wuisawesome — APPROVED** [review]

**Comment by DmitriGekhtman** [comment]
Windows and RLLib failures unrelated.

**Pull request merged**